### PR TITLE
feat: modify the default value of Attribute.Component Repeatable

### PR DESCRIPTION
**Pull request by Convly**
### What does it do?

- Modify the default value of the `TRepeatable` type param for `Attribute.Component`, from `false` to `boolean`
- Update `Attribute.Component` value resolution to take into account a potential `true | false` (`boolean`) value for `TRepeatable`
- Update type generation
- Update tests

### Why is it needed?

- `Attribute.Component` `TRepeatable` resolving to `false` by default was frustrating when using an `Attribute.Component` as it made type narrowing harder (the value couldn't be considered as `true`, only `false` or `undefined`, the solution was then to use  `Attribute.Component<Common.UID.Schema, boolean>`, but wasn't very DX friendly

### How to test it?

- Make sure `Attribute.Component` (as well as other types using it such as `Attribute.GetValue`) are still working as they should
- The unit test should pass

### CODE REVIEW

**Comment by Convly** [comment]
For a mysterious reason, something is broken in the admin. Tried to reproduce locally but it seems the admin cannot even access the Attribute.GetValue :thinking: 

I'll investigate tomorrow

File: packages/core/types/src/types/core/attributes/component.ts [hunk 1]
```diff
@@ -27,7 +27,16 @@ export type ComponentValue<
...
-  ? Utils.Expression.If<TRepeatable, TValues[], TValues>
+  ? Utils.Expression.MatchFirst<
+      [
+        // Repeatable component
+        [Utils.Expression.IsTrue<TRepeatable>, TValues[]],
+        // Single component
+        [Utils.Expression.IsFalse<TRepeatable>, TValues]
+      ],
+      // If TRepeatable is neither true nor false, then return every possible types
+      TValues[] | TValues
```
Comment (alexandrebodin) [inline packages/core/types/src/types/core/attributes/component.ts:38]: 🤔  in the case the repeatable field is not set in the schema we still set it to false so can we make it required so we don't need that fallback that doesn't mean anything ?

Comment (Convly) [reply packages/core/types/src/types/core/attributes/component.ts:38]: Here it's more about "I don't know wether it's false or true, not if this is undefined

Comment (Convly) [reply packages/core/types/src/types/core/attributes/component.ts:38]: In short: `Attribute.Component` is a component that can be repeatable or not (`true | false` -> `boolean`), by making the check on `.repeatable`, you should be able to infer the correct values

Comment (alexandrebodin) [reply packages/core/types/src/types/core/attributes/component.ts:38]: but this should always be one or the other (we are missing a `is` operator somewhere :D) but in that case I would return a never instead to represent there is not valid return type 🤷

**Review by innerdvations — APPROVED** [review]
LGTM, we just have the failing e2e tests on v5/main that need to be fixed before we can be sure.

**Pull request merged**